main.go

```diff
@@ -36,165 +48,227 @@ var (
+				continue
 			}
 
+			if result != chrysom.CreatedPushResult && result != chrysom.UpdatedPushResult {
+				logging.Error(logger).Log(logging.MessageKey(), "Unsuccessful item push response from Argus", "err", err)
+			}
+			logging.Debug(logger).Log(logging.MessageKey(), "Successfully pushed an webhook item from SNS to Argus")
 		}
 	}
 }
 
-func printVersion(f *pflag.FlagSet, arguments []string) (error, bool) {
-	printVer := f.BoolP("version", "v", false, "displays the version number")
-	if err := f.Parse(arguments); err != nil {
-		return err, true
+func webhookToItem(w *webhook.W) (*model.Item, error) {
+	encodedWebhook, err := json.Marshal(w)
+	if err != nil {
+		return nil, err
 	}
-
-	if *printVer {
-		printVersionInfo(os.Stdout)
-		return nil, true
+	var data map[string]interface{}
+	err = json.Unmarshal(encodedWebhook, &data)
+	if err != nil {
+		return nil, err
 	}
-	return nil, false
+
+	checkSumURL := sha256.Sum256([]byte(w.Config.URL))
+	TTLSeconds := int64(w.Duration.Seconds())
+
+	return &model.Item{
+		Data: data,
+		ID:   base64.RawURLEncoding.EncodeToString(checkSumURL[:]),
+		TTL:  &TTLSeconds,
+	}, nil
 }
 
-func printVersionInfo(writer io.Writer) {
-	fmt.Fprintf(writer, "%s:\n", applicationName)
-	fmt.Fprintf(writer, "  version: \t%s\n", Version)
-	fmt.Fprintf(writer, "  go version: \t%s\n", runtime.Version())
-	fmt.Fprintf(writer, "  built time: \t%s\n", BuildTime)
-	fmt.Fprintf(writer, "  git commit: \t%s\n", GitCommit)
-	fmt.Fprintf(writer, "  os/arch: \t%s/%s\n", runtime.GOOS, runtime.GOARCH)
+func printVersionInfo() {
+	fmt.Fprintf(os.Stdout, "%s:\n", applicationName)
+	fmt.Fprintf(os.Stdout, "  version: \t%s\n", Version)
+	fmt.Fprintf(os.Stdout, "  go version: \t%s\n", runtime.Version())
+	fmt.Fprintf(os.Stdout, "  built time: \t%s\n", BuildTime)
+	fmt.Fprintf(os.Stdout, "  git commit: \t%s\n", GitCommit)
+	fmt.Fprintf(os.Stdout, "  os/arch: \t%s/%s\n", runtime.GOOS, runtime.GOARCH)
 }
 
-func exitIfError(logger log.Logger, err error) {
-	if err != nil {
-		if logger != nil {
-			logging.Error(logger, emperror.Context(err)...).Log(logging.ErrorKey(), err.Error())
-		}
-		fmt.Fprintf(os.Stderr, "Error: %#v\n", err.Error())
-		os.Exit(1)
-	}
+func ApplyMetricsData(awsMetrics aws.AWSMetrics, webhookMetrics webhook.WebhookMetrics) {
+	webhookMetrics.ListSize.Add(0.0)
+	webhookMetrics.NotificationUnmarshallFailed.Add(0.0)
+
+	awsMetrics.DnsReadyQueryCount.Add(0.0)
+	awsMetrics.DnsReady.Add(0.0)
+	awsMetrics.SNSNotificationSent.Add(0.0)
+	awsMetrics.SNSSubscribed.Add(0.0)
 }
 
-func toNewWebhook(w *webhook.W) (*xwebhook.Webhook, error) {
-	data, err := json.Marshal(w)
-	if err != nil {
-		return nil, err
-	}
-	xw := new(xwebhook.Webhook)
-	err = json.Unmarshal(data, xw)
-	if err != nil {
-		return nil, err
-	}
-	return xw, nil
+// TODO: once we get rid of any packages that need an unmarshaller, remove this.
+type UnmarshallerOut struct {
```

Comment:
We can remove this by using arrange https://pkg.go.dev/github.com/xmidt-org/arrange#hdr-Unmarshaled_components
and so we can get rid of this struct 

From the example code, it would look like:
```
v := viper.New()
fx.New(
  arrange.FromViper(v),
  arrange.ProvideKey("argus", *chrysom.ClientConfig{}), 
  fx.Invoke(
    func(cfg *chrysom.ClientConfig) error {
      // use the Config as desired
    },
  ),
)
```

Reply:
I think this struct is needed for the xhealth unmarshal [here](https://github.com/xmidt-org/themis/blob/00676a46dcb1249c9cf9f6cd47f4b977fbaf3ec8/xhealth/unmarshal.go#L48), as the HealthIn struct needs a config.Unmarshaller: https://github.com/xmidt-org/themis/blob/00676a46dcb1249c9cf9f6cd47f4b977fbaf3ec8/xhealth/unmarshal.go#L21 ?

Reply:
I thought we could use https://github.com/xmidt-org/themis/blob/00676a46dcb1249c9cf9f6cd47f4b977fbaf3ec8/config/viper.go#L50 but @johnabass just told me themis and arrange differ a bit on how the viper/unmarshaling is linked/setup. Sounds like he's working on a new arrange release that might help. Let's go with this for now 🚀